### PR TITLE
CORE-6564 Add functionality to allow counterparty flow information to  be exchanged

### PR DESCRIPTION
**Pull request by LWogan**
Exchange a map of properties as part of session initiation. 
move the protocol info into the session properties. 
Include a new SessionConfirm message which will be sent in response to SessionInits to pass the session properties from initiated to initiating sessions. 
add a waiting for handler to wait for session confirmation before trying to return FlowInfo to the Fiber.

### CODE REVIEW

**Review by mbrkic-r3 — DISMISSED** [review]
LGTM

File: application/src/main/java/net/corda/v5/application/messaging/FlowInfo.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+public interface FlowInfo {
+    /**
+     * @return The protocol name the flow is running.
+     */
+    String protocol();
+
+    /**
+     * @return The protocol version the flow is running.
+     */
+    Integer protocolVersion();
```
Comment (mbrkic-r3) [inline application/src/main/java/net/corda/v5/application/messaging/FlowInfo.java:18]: Is this mandatory?

Comment (LWogan) [reply application/src/main/java/net/corda/v5/application/messaging/FlowInfo.java:18]: yes. added NotNull annotation

**Review by mbrkic-r3 — APPROVED** [review]
LGTM

**Review by mnesbit — APPROVED** [review]
OK, but I will discuss with @JamesHR3 about clarifying role with regard to existing flow properties API

**Pull request merged**